lib/internal/worker.js

```diff
@@ -445,6 +445,9 @@ function setupChild(evalScript) {
 
   function fatalException(error) {
     debug(`[${threadId}] gets fatal exception`);
+
+    process.exitCode = 1;
```

Comment:
I'm afraid process.exitCode may be 1 even if the exception is caught.

Reply:
I agree – I think you want to move this into the `if (!caught)` block below?

Reply:
My bad, I assumed that this was unrecoverable (I thought 'uncaughtException' didn't count as recovery). Will push soon. I also added test similar to `test-process-exit-code` for workers.

Reply:
I think behaviour should match the main thread’s behaviour as closely as possible – in there, we have that `process.exitCode` is not set, as far as I can tell:

```js
process.on("uncaughtException", () => console.log(process.exitCode));
throw new Error();11
```

prints `undefined`.